### PR TITLE
RED-611 Moved common logic out and accounted for optional

### PR DESCRIPTION
**Pull request by red-tt-bear**

### CODE REVIEW

File: src/features/courses/course-controller.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ import curriculumRepository from '../curriculum/curriculum-repository';
 import CurriculumUnitContent from '../../database/models/curriculum-unit-content';
 import CurriculumTopicContent from '../../database/models/curriculum-topic-content';
 import CurriculumWWTopicQuestion from '../../database/models/curriculum-ww-topic-question';
-import WebWorkDef, { Problem } from '@rederly/webwork-def-parser';
+import WebWorkDef, { Problem } from '@rederly/rederly-utils/lib/webwork-def-parser';
```
Comment (PanopticaRising) [inline src/features/courses/course-controller.ts:28]: It seems a bit unintuitive to have to pull this out from a subdirectory of an npm package.

File: src/features/courses/course-controller.ts [hunk 2]
```diff
@@ -1641,7 +1641,7 @@ class CourseController {
                             weight: parseInt(problem.value ?? '1'),
                             maxAttempts: parseInt(problem.max_attempts ?? '-1'),
                             hidden: false,
-                            optional: false,
+                            optional: problem.rederlyOptional === 'true'
```
Comment (PanopticaRising) [inline src/features/courses/course-controller.ts:1644]: Personally, I like doing `true.toString()` to be consistent

**Review by PanopticaRising — APPROVED** [review]

File: src/features/courses/course-controller.ts (hunk 1 above)
Comment (red-tt-bear) [reply src/features/courses/course-controller.ts:28]: Yeah, I wanted to keep the exports the same for now, the default export itself is probably the most annoying part

In the future I might refactor

File: src/features/courses/course-controller.ts (hunk 2 above)
Comment (red-tt-bear) [reply src/features/courses/course-controller.ts:1644]: Coming from the guy that does `++i` for efficiency 🤣

**Pull request merged**